genie-agent/src/main/java/com/netflix/genie/agent/execution/locks/impl/LockingFactory.java

```diff
@@ -0,0 +1,55 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.agent.execution.locks.impl;
+
+import com.netflix.genie.agent.execution.locks.Lock;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+import java.nio.channels.FileChannel;
+
+/**
+ * Factory for creating locks implementing {@link com.netflix.genie.agent.execution.locks.Lock}.
+ *
+ * @author standon
+ * @since 4.0.0
+ */
+
+@Slf4j
+@Component
+@Lazy
+public class LockingFactory {
+
+    /**
+     * Get a lock backed by the passed in object.
+     * @param objectBackingLock object used to derive the lock
+     * @return a lock backed by the passed in object
+     */
+    public Lock getLock(final Object objectBackingLock) {
```

Comment:
This seems pretty roundabout at least at first glance to me. You're exposing an internal mechanism (FileChannel) of your locking implementation. It seems to me your clients to your Locking logic shouldn't know or care how this works and shouldn't really even know about FileChannels and whatnot. They should just pass in a `File` or a `URL` or something and say "lock this". The way it's done should be hidden from the outside. In this way you're asking clients to send in a `FileChannel` instead which isn't an abstract construct it's pretty concrete. At that point they may as well construct a `FileLockBackedLock` themselves in place rather than calling this factory. Ideally I'd think if they passed in a `File` or a `URL` or a `Resource` the internal logic here would determine the best implementation to use and return it...